python_modules/dagster/dagster/_utils/caching_instance_queryer.py

```diff
@@ -707,9 +707,12 @@ def asset_partitions_with_newly_updated_parents_and_new_cursor(
                         f"and child {child_asset_key.to_string()}."
                     ) from e
 
-                child_partitions = reversed(list(child_partitions_subset.get_partition_keys()))
+                child_partitions = sorted(
```

Comment:
maybe add a comment here explaining that this is more of a "best effort" sorting, as there's no guarantee that lexographically sorting arbitrarily-formatted datetimes will actually result in date sorting